src/ngraph/frontend/onnxifi/backend_manager.cpp

```diff
@@ -0,0 +1,67 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <cstdlib>   // std::size_t, std::uintptr_t
+#include <stdexcept> // std::invalid_agrument, std::out_of_rage
+
+#include <onnxifi.h>
+
+#include "ngraph/runtime/backend_manager.hpp"
+
+#include "backend.hpp"
+#include "backend_manager.hpp"
+
+namespace ngraph
+{
+    namespace onnxifi
+    {
+        BackendManager::BackendManager()
+        {
+            auto registered_backends = runtime::BackendManager::get_registered_backends();
```

Comment:
I'm trying but I don't understand what this class is going. registered_backends is a vector<string> and then you are casting the address to this temporary string to a uintptr_t on line 36. This looks wrong.

Reply:
`registered_backends` is used to eliminate the potential _undefined behavior_ which may occur if one uses the result of a function (which is `vector`) as `range_expression` of the range-based loop. To resolve this issue future standard (i.e. C++20) will provide `init-statement` section of the range-based loop (see https://en.cppreference.com/w/cpp/language/for for details).

Reply:
`m_registered_backends` is the `std::map` which is mapping **handles** to `onnxifi::Backend` **objects**. The backend uses pointer (via `std::uintptr_t` because by the specification its size is equal size of pointer) to `type` as handle to `ngraph::runtime::Backend`.
So it does not matter whether `type` is temporar or permament becuaus it is used only as **handle** (a reference to actual object). The lifetime of `onnxifi::BackendManager` is determined by the lifetime of application (singleton pattern).
Accessing objects through handles is mandatory requirement of ONNXIFI specification.

Reply:
So add a comment say that. You can't just write opaque code without comment.